core/src/main/java/org/apache/calcite/rel/rules/UnionEliminatorRule.java

```diff
@@ -44,18 +46,17 @@ public UnionEliminatorRule(Class<? extends Union> clazz,
 
   //~ Methods ----------------------------------------------------------------
 
-  public void onMatch(RelOptRuleCall call) {
+  @Override public boolean matches(RelOptRuleCall call) {
     Union union = call.rel(0);
-    if (union.getInputs().size() != 1) {
-      return;
-    }
-    if (!union.all) {
-      return;
-    }
-
-    // REVIEW jvs 14-Mar-2006:  why don't we need to register
-    // the equivalence here like we do in AggregateRemoveRule?
+    return union.all && union.getInputs().size() == 1;
```

Comment:
I am curious when can we get a union with only 1 child?

Reply:
I guess one possible case is this. Unions are typically created via:
public static LogicalUnion create(List<RelNode> inputs, boolean all);

"List<RelNode> inputs" is typically assembled with a for loop. Whenever the for loop executes with only one loop, we have an input list of one element.